server/services/status-history-logger.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+var resolveProcessingStepFullPath = require('../../utils/resolve-processing-step-full-path')
+  , serializeValue                = require('dbjs/_setup/serialize/value')
+  , unserializeValue              = require('dbjs/_setup/unserialize/value')
+  , capitalize                    = require('es5-ext/string/#/capitalize')
+  , debug                         = require('debug-ext')('status-history-logger')
+  , db                            = require('../../db')
+  , Set                           = require('es6-set')
+  , processingStepsMeta           = require('../../processing-steps-meta')
+  , uuid                          = require('time-uuid')
+  , uniqIdPrefix                  = 'abcdefghiklmnopqrstuvxyz'[Math.floor(Math.random() * 24)];
+
+var getPathSuffix = function () {
+	return 'statusHistory/map/' + uniqIdPrefix + uuid();
+};
+
+var storeLog = function (storage, logPath/*, options */) {
+	var options = Object(arguments[2]);
+	return function (event) {
+		var status, oldStatus, resolvedPath, logPathResolved;
+		logPathResolved = logPath ? logPath + '/' + getPathSuffix() : getPathSuffix();
+
+		if (event.type !== 'computed') return;
+		status    = unserializeValue(event.data.value) || null;
+		oldStatus = (event.old && unserializeValue(event.old.value)) || null;
```

Comment:
As we discussed let's not store `null` but `''` (`null` is not acceptable value for `cardinalPropertyKey`)